zenoh-flow/src/io/output.rs

```diff
@@ -190,14 +190,14 @@ impl OutputBuilder {
     /// ```
     pub fn build_typed<T: Send + Sync + 'static>(
         self,
-        serializer: impl Fn(&T) -> anyhow::Result<Vec<u8>> + Send + Sync + 'static,
+        serializer: impl Fn(&mut Vec<u8>, &T) -> anyhow::Result<()> + Send + Sync + 'static,
```

Comment:
Why not a slice `&mut [u8]` ?

If it is because it cannot grow then I'm fine with it.